lib/Parse/ParseDecl.cpp

```diff
@@ -6044,6 +6059,18 @@ void Parser::ParseParameterDeclarationClause(
 
     // If the next token is a comma, consume it and keep reading arguments.
   } while (TryConsumeToken(tok::comma));
+
+  // Now parse the deferred bounds expressions
+  for (auto Pair : deferredBoundsExpressions) {
```

Comment:
>auto [](start = 7, length = 4)

Did you mean to use an 'auto &' instead?  Or better yet, perhaps a 'const auto &'?  That avoids copies and makes sure you're not mutating the source list.